### PR TITLE
BAU: replace charge_id with transaction_id

### PR DESCRIPTION
**Pull request by jankowiakmaria**
* for transaction search endpoint
* in order to deprecate charge_id in ledger

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import com.google.common.collect.ImmutableSet;
 import com.google.inject.Inject;
+import org.skife.jdbi.v2.sqlobject.Transaction;
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java]: don't think this is what you wanted?

File: src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java [hunk 2]
```diff
@@ -87,12 +89,14 @@ public TransactionSearchResults doSearch(Account account, TransactionSearchParam
             return processResponse(ledgerResponse);
         }
         throw new SearchTransactionsException(ledgerResponse);
+
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java]: unnecessary line

File: src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java [hunk 3]
```diff
@@ -111,14 +115,14 @@ private TransactionSearchResults processResponse(Response connectorResponse) {
         );
     }
 
-    private PaymentForSearchResult getPaymentForSearchResult(ChargeFromResponse charge) {
+    private PaymentForSearchResult getPaymentForSearchResult(TransactionResponse charge) {
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java]: maybe rename `charge` to `transaction` or `transactionResponse`?

File: src/main/java/uk/gov/pay/api/service/PaymentSearchService.java [hunk 4]
```diff
@@ -99,9 +101,9 @@ private Response getSearchResponse(Account account, Map<String, String> queryPar
     }
 
     private Response processResponse(Response connectorResponse) {
-        PaymentSearchResponse response;
+        PaymentSearchResponse<ChargeFromResponse> response;
         try {
-            response = connectorResponse.readEntity(PaymentSearchResponse.class);
+            response = connectorResponse.readEntity(new GenericType<PaymentSearchResponse<ChargeFromResponse>>(){});
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/api/service/PaymentSearchService.java]: this is pretty nifty

File: src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java (hunk 1 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java]: oops, alt + enter too fast (sometime IntelliJ is not smart enough).

File: src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java (hunk 3 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/ledger/service/TransactionSearchService.java]: fair enough

File: src/main/java/uk/gov/pay/api/service/PaymentSearchService.java (hunk 4 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/service/PaymentSearchService.java]: I know, reads quite nicely.

**Pull request merged**